utilities/write_batch_with_index/write_batch_with_index.cc

```diff
@@ -264,32 +276,38 @@ class BaseDeltaIterator : public Iterator {
       } else if (!delta_iterator_->status().ok()) {
         // Expose the error status and stop.
         current_at_base_ = false;
-        return;
+        break;
```

Comment:
It's safer to just return here. I am worried about by mistake we access the data while status is not OK. It always makes things hard to reason about.